docs/madr/decisions/056-meshservice-migration.md

```diff
@@ -0,0 +1,343 @@
+# MeshService migration
+
+- Status: accepted
+
+## Context and Problem Statement
+
+This MADR addresses the switch from not having MeshServices at all to MeshServices
+being automatically created, either from `Services` in Kubernetes or from
+`Dataplanes` in Universal.
+
+When upgrading from 2.8 to 2.9, nothing at the data plane level should change.
+But MeshService behaves differently in several ways.
+
+In general the migration process for MeshService consists of no longer using
+`kuma.io/service` so:
+
+- targeting policies to the real `MeshService` resource
+- switching over to the `MeshService` i.e. `HostnameGenerator` hostnames and
+  local-zone load balancing from cross-zone load balancing
+
+Ideally users can try out `MeshService` behavior for individual consumers.
+
+Note that we in general can't be smart about which behavior should be the
+default because we can't differentiate between a new user, who doesn't care
+about `kuma.io/service`-style behavior, and an existing user who's just
+redeploying on a new `Mesh` or control plane.
+
+We have to make a choice between:
+
+- new behavior by default, `UPGRADE.md` tells users to set a variable to keep
+  old behavior
+- old behavior by default, new users have to be convinced to set a variable to
+  get new behavior
+
+### `MeshService`-targeted policy
```

Comment:
Probably worth mentioning `backendRefs` inside Mesh*Route policies. Do we want to convert existing one to targetRef with labels as well?

Reply:
Yeah we do.

Reply:
Actually this doesn't quite make sense like it does for `targetRef`. It's important that these are manually updated because after MeshServices are created, it isn't clear whether traffic should go to that specific MeshService or maybe a MeshMultiZoneService.

Reply:
If we accept the above I'll add it to the MADR as part of the migration.

Reply:
But if you have a TCP route like:
```yaml
to:
  - targetRef:
      kind: MeshService
      name: backend_demo-ns_svc_8080
    rules:
      - default:
          backendRefs:
            - targetRef:
                kind: MeshService
                name: backend-v2_demo-ns_svc_8080
```
don't you expect after MeshServices are generated the route still works?